d3log/src/tcp_network.rs

```diff
@@ -45,8 +45,8 @@ pub fn nid_to_socketaddr(n:Node) -> SocketAddr {
     SocketAddr::new(IpAddr::V4(Ipv4Addr::new(b[0], b[1], b[2], b[3])), u16::from_be_bytes([b[4], b[5]]))
 }
 
-pub struct TcpNetwork {
-    peers : Arc::<Mutex::<HashMap<Node, Arc::<Mutex::<TcpStream>>>>>
+pub struct TcpNetwork<'a> {
+    peers : &'a Arc::<Mutex::<HashMap<Node, Arc::<Mutex::<TcpStream>>>>>
```

Comment:
What's the referencing for here? You shouldn't need lifetimes at all (much less getting a static one), you can just clone the `Arc` and that'll keep the data alive as long as references are outstanding

Reply:
Why do we need to wrap this in `Arc<Mutex>` given that we wrap the entire `TcpNetwork` into that below?

Reply:
static and lifetimes are gone. I was trying to line things up so that lifetimes were static inside an async block so that the compiler would mark it send(). I got away with using clones of arcs.